### PR TITLE
dockershim: remove the use of kubelet's internal API

### PR DESCRIPTION
**Pull request by yujuhong**
We let dockershim implement the kubelet's internal (CRI) API as an
intermediary step before transitioning fully to communicate using gRPC.
Now that kubelet has been communicating to the runtime over gRPC for
multiple releases, we can safely retire the extra interface in
dockershim.

This PR also moves the legacy functions to a separate file and clean up
the interfaces.

### CODE REVIEW

**Comment by yujuhong** [comment]
I removed `dockershim/remote` from `hack/.golint_failures`. As a result of this, I'd need the approval from an owner of  the `hack` package.
/assign @ixdy 
help?

**Comment by Random-Liu** [comment]
LGTM with one nit

File: pkg/kubelet/dockershim/docker_legacy_service.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	// IsCRISupportedLogDriver checks whether the logging driver used by docker is
+	// suppoted by native CRI integration.
+	// TODO(resouer): remove this when deprecating unsupported log driver
+	IsCRISupportedLogDriver() (bool, error)
+
+	kuberuntime.LegacyLogProvider
+}
+
+// NewDockerLegacyService created docker legacy service when log driver is not supported.
```
[the remark this answers is not in the document]
Comment (yujuhong) [reply pkg/kubelet/dockershim/docker_legacy_service.go]: Done.

**Comment by yujuhong** [comment]
Thanks a lot for the review!

**Comment by yujuhong** [comment]
Applied lgtm based on https://github.com/kubernetes/kubernetes/pull/58548#issuecomment-359131391

**Pull request merged**